testsuite/MDAnalysisTests/analysis/test_bat.py

```diff
@@ -127,3 +128,16 @@ def test_bat_incorrect_dims(self, bat_npz):
         errmsg = 'Dimensions of array in loaded file'
         with pytest.raises(ValueError, match=errmsg):
             R = BAT(selected_residues, filename=bat_npz)
+
+    def test_bat_transformation(self, selected_residues, bat):
+        R = BAT(selected_residues)
+        R.run()
+        R.Cartesian(R.results.bat[0])
+        XYZ = R.results.bat[0]
+        PQR = copy.deepcopy(XYZ)
+        R.Cartesian(R.results.bat[0])
+        assert_almost_equal(PQR, XYZ, 5,
+                            err_msg="error: Reconstructed Cartesian"
+                                    "coordinate after multiple "
+                                    "transformations don't match "
+                                    "original")
```

Comment:
Could you please explain this test to me? Which lines test the specific problem in #3501 ? Is `R.run()` necessary if there's already a `bat` fixture? Why have you chosen to call the coordinates XYZ and PQR -- can you think of more informative variable names? Why do you run `R.Cartesian` twice? Does the error message still make sense for the assertion?

Keep in mind that the specific problem in #3501 is (quoting from the first comment):

> When calling the Cartesian method of a BAT class instance to convert BAT coordinates back to Cartesian, the input BAT data should not be modified.

Reply:
If we do not use R.run() then we get the errors that Results object has no attribute bat . Yes should I call XYZ as bat coords and PQR as deepcopy of  original bat co-ords.
No it wasn't necessary to run Cartesian twice. I changed the error message .

Reply:
According to my latest commit I store the original bat coords in XYZ and create a deepcopy of them in PQR. Then after running the cartesian method the bat coords(XYZ) get modified  and so aren't equal to PQR .